README.rst
lizard-geo
==========================================

Introduction
------------

Store geo objects and manage imports. The app provides the basics for
storing geographical entities. GeoObjects store the entities
themselves and GeoObjectGroup groups the GeoObjects.

See also lizard-area and lizard-fewsnorm as examples.


Development installation
------------------------

The first time, you'll have to run the bootstrap script to set up setuptools
and buildout::

    $> python bootstrap.py

And then run buildout to set everything up::

    $> bin/buildout

Note that on Microsoft Windows it is called ``bin\buildout.exe``.

You will have to re-run buildout when ``setup.py`` or ``buildout.cfg`` have
been modified, for example directly by you or through an update of your working
directory.

The current package is installed as a "development package", so changes in .py
files are automatically available (just like with ``python setup.py develop``).

The app lizard-area uses a SpatiaLite database, although this will probably be
overwritten at the site level. The repository already contains a SpatiaLite
database, so you should not need to create one yourself. In case you want to
create a new SpatiaLite database, execute the following command::

  $> spatialite lizard-area.db < init_spatialite-2.3.sql
  spatialite lizard-area.db < init_spatialite-2.3.sql
  SpatiaLite version ..: 2.3.0	Supported Extensions:
        - 'VirtualShape'        [direct Shapefile access]
        - 'VirtualText          [direct CSV/TXT access]
        - 'VirtualNetwork       [Dijkstra shortest path]
        - 'RTree'               [Spatial Index - R*Tree]
        - 'MbrCache'            [Spatial Index - MBR cache]
        - 'VirtualFDO'          [FDO-OGR interoperability]
        - 'SpatiaLite'          [Spatial SQL - OGC]
  PROJ.4 version ......: Rel. 4.7.1, 23 September 2009
  GEOS version ........: 3.1.0-CAPI-1.5.0

The above snippet assumes you already have installed SpatiaLite.

To create the database tables and fill them do::

  $ bin/django syncdb

.. _South: http://south.aeracode.org/

Several lizard packages use South_ to handle the creation of tables. The above
command will inform you which packages have to be handled by South. For
example, its output will contains messages such as these::

  Not synced (use migrations):
   - lizard_geo
   - lizard_ui
  (use ./manage.py migrate to migrate these)

You can create the tables for these packages, or using the South terminology,
'migrate' them, as follows::

  $ bin/django migrate


Usage
-----

Most of the time you want to extend GeoObject because extra fields are
needed. When importing from a datasource, for instance a shape file,
you can make a GeoObjectGroup for each import. The purpose for a
GeoObjectGroup is that you can remove GeoObjects later by removing the
GeoObjectGroups.

Note that you must use PostGIS 1.5 or newer in order to make spatial
queries on the GeometryField. A spatial queryset is for example:
GeoObject.objects.filter(geometry__distance_lte=(pnt, D(m=10))


CHANGES.rst
Changelog of lizard-geo
===================================================


0.4 (unreleased)
----------------

- Nothing changed yet.


0.3 (2011-11-29)
----------------

- Removed unique from GeoObject.ident. The administrator himself
  should manage the uniqueness of idents.


0.2 (2011-11-07)
----------------

- Pylint.

- Updated readme.

- Changed max_length of GeoObject.ident from 16 to 80.


0.1 (2011-10-19)
----------------

- Moved models from lizard-area.

- Initial library skeleton created by nensskel.  [Jack Ha]


lizard_geo/migrations/0002_auto__chg_field_geoobject_ident.py
# encoding: utf-8
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models

class Migration(SchemaMigration):

    def forwards(self, orm):
        
        # Changing field 'GeoObject.ident'
        db.alter_column('lizard_geo_geoobject', 'ident', self.gf('django.db.models.fields.CharField')(unique=True, max_length=80))


    def backwards(self, orm):
        
        # Changing field 'GeoObject.ident'
        db.alter_column('lizard_geo_geoobject', 'ident', self.gf('django.db.models.fields.CharField')(max_length=16, unique=True))


    models = {
        'auth.group': {
            'Meta': {'object_name': 'Group'},
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '80'}),
            'permissions': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Permission']", 'symmetrical': 'False', 'blank': 'True'})
        },
        'auth.permission': {
            'Meta': {'ordering': "('content_type__app_label', 'content_type__model', 'codename')", 'unique_together': "(('content_type', 'codename'),)", 'object_name': 'Permission'},
            'codename': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'content_type': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['contenttypes.ContentType']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '50'})
        },
        'auth.user': {
            'Meta': {'object_name': 'User'},
            'date_joined': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'email': ('django.db.models.fields.EmailField', [], {'max_length': '75', 'blank': 'True'}),
            'first_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'groups': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Group']", 'symmetrical': 'False', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'is_active': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'is_staff': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'is_superuser': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'last_login': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'last_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'password': ('django.db.models.fields.CharField', [], {'max_length': '128'}),
            'user_permissions': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Permission']", 'symmetrical': 'False', 'blank': 'True'}),
            'username': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '30'})
        },
        'contenttypes.contenttype': {
            'Meta': {'ordering': "('name',)", 'unique_together': "(('app_label', 'model'),)", 'object_name': 'ContentType', 'db_table': "'django_content_type'"},
            'app_label': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'model': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
        },
        'lizard_geo.geoobject': {
            'Meta': {'object_name': 'GeoObject'},
            'geo_object_group': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['lizard_geo.GeoObjectGroup']"}),
            'geometry': ('django.contrib.gis.db.models.fields.GeometryField', [], {}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'ident': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '80'})
        },
        'lizard_geo.geoobjectgroup': {
            'Meta': {'object_name': 'GeoObjectGroup'},
            'created_by': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['auth.User']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'last_modified': ('django.db.models.fields.DateTimeField', [], {'auto_now': 'True', 'blank': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '128'}),
            'slug': ('django.db.models.fields.SlugField', [], {'unique': 'True', 'max_length': '50', 'db_index': 'True'}),
            'source_log': ('django.db.models.fields.TextField', [], {'null': 'True', 'blank': 'True'})
        }
    }

    complete_apps = ['lizard_geo']
